### PR TITLE
Adding perl bindings

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Been struggling to get perl/sysroot to behave correctly so we'll see how this goes.

Once it's compiling correctly the intention is to rework this recipe to generate two outputs, `subversion` and `subversion-perl`.

### CODE REVIEW

**Comment by scopatz** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ requirements:
     - nasm
     - make
     - python >=2.7,<3
+    - perl >=5.8.0
```
Comment (scopatz) [inline recipe/meta.yaml]: It seems like perl is maybe a host dep, if it is also a run dep

Comment (kenodegard) [reply recipe/meta.yaml]: That may definitely be the case, still haven't fully understood the host vs build distinction

Comment (scopatz) [reply recipe/meta.yaml]: You can think of it as a cross-compiler distinction.  `build` is the architecture of the machine you are actually building on (e.g. linux-64), where as `host` is the target architecture that you are building for and so it is arch where dynamic libraries are needed to link against (e.g. osx-64). 

So compilers and build tools tend to go in build, while libraries that the code uses go into host.

Comment (scopatz) [reply recipe/meta.yaml]: perl is in a werid place because it is often used as a build tool, but is also a target to be linked against

Comment (kenodegard) [reply recipe/meta.yaml]: Ah, I was getting held up on the Python example which is being used as a build tool. I need to think of Perl as being a lib that I'm linking against.

**Comment by kenodegard** [comment]
@scopatz I've found that the `sys/types.h` error is due to the **swig-pl** process not finding the correct **sysroot**.

`configure` does the following:
```bash
SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts`"
```

I expect it to be using this **sysroot** path:
```
${PREFIX}/x86_64-conda-linux-gnu/sysroot
```

But if I run `$PERL -MExtUtils::Embed -e ccopts` I end up getting this **sysroot** path which doesn't exist:
```
${PREFIX}/bin/..//x86_64-conda_cos6-linux-gnu/sysroot
```

I've been able to circumvent this issue locally by symlinking `${PREFIX}/x86_64-conda-linux-gnu/sysroot` into `${PREFIX}/bin/..//x86_64-conda_cos6-linux-gnu/sysroot` however that feels like the wrong approach to take here.

Do you have any thoughts on this?

**Comment by scopatz** [comment]
sysroot is handled somewhat specially. Let's ask @isuruf & @beckermr

**Comment by beckermr** [comment]
Maybe rebuild perl?

**Comment by scopatz** [comment]
CC @conda-forge/perl

**Comment by beckermr** [comment]
I am not 100% sure, but if perl is vendoring a copy of the sysroot path then it will be looking for the old one.

**Comment by kenodegard** [comment]
Submitted conda-forge/perl-feedstock#42 to rebuild perl.

**Comment by zklaus** [comment]
This would help resolve conda-forge/git-feedstock#60 by providing the necessary perl bindings, so I support this effort! Should there be an issue corresponding to this PR?

**Comment by zklaus** [comment]
[Subversions swig install instructions](https://svn.apache.org/repos/asf/subversion/trunk/subversion/bindings/swig/INSTALL) suggest that we should pass something like `--with-swig-perl=$CONDA_PREFIX/bin/perl`. Maybe that could resolve the sysroot issue?

**Comment by kenodegard** [comment]
Closing in favor of #7

**Pull request closed without merging**